.github/workflows/ci.yml

```diff
@@ -0,0 +1,443 @@
+name: Build and test gem
+
+on:
+  push:
```

Comment:
We _might_ want to limit push to master, keeping the rest of the on's as you have it. Another way to say it, I don't think pushes to branches w/o PR's needs to run this CI workflow...